services/libs/aster-std/src/syscall/mmap.rs

```diff
@@ -78,17 +78,29 @@ fn mmap_anonymous_vmo(
         panic!("Unsupported mmap flags {:?} now", option);
     }
 
-    let vmo_options: VmoOptions<Rights> = VmoOptions::new(len);
-    let vmo = vmo_options.alloc()?;
-    let current = current!();
-    let root_vmar = current.root_vmar();
+    let vmo = {
+        let vmo_options: VmoOptions<Rights> = VmoOptions::new(len);
+        vmo_options.alloc()?
+    };
+
+    let mut vmar_map_options = {
+        let current = current!();
+        let root_vmar = current.root_vmar();
+        root_vmar.new_map(vmo, perms)?
+    };
 
-    let mut vmar_map_options = root_vmar.new_map(vmo, perms)?;
-    if option.flags().contains(MMapFlags::MAP_FIXED) {
+    let flags = option.flags();
+    if flags.contains(MMapFlags::MAP_FIXED) {
         vmar_map_options = vmar_map_options.offset(addr).can_overwrite(true);
+    } else if flags.contains(MMapFlags::MAP_32BIT) {
+        // TODO: support MAP_32BIT. MAP_32BIT requires the map range to be below 2GB
+        warn!("MAP_32BIT is not supported");
     }
```

Comment:
The 6 lines above are duplicated in both `mmap_anonymous_vmo` and `mmap_filebacked_vmo`.